auth_token/models.py

```diff
@@ -129,55 +171,357 @@ def delete(self):
         raise NotImplementedError
 
 
-class DeviceKeyAlreadyExistsException(Exception):
+class UserAuthorizationTokenTakeover(SmartModel):
+    """
+    The model allows to change user without token change
+    """
+
+    token = models.ForeignKey(
+        verbose_name=_('authorization token'),
+        to=AuthorizationToken,
+        related_name='user_takeovers',
+        on_delete=models.CASCADE
+    )
+    user = models.ForeignKey(
+        verbose_name=_('user'),
+        to=django_settings.AUTH_USER_MODEL,
+        related_name='user_token_takeovers',
+        null=False,
+        blank=False,
+        on_delete=models.CASCADE
+    )
+    is_active = models.BooleanField()
+
+    class Meta:
+        verbose_name = _('authorization takeover')
+        verbose_name_plural = _('authorization takeovers')
+
+
+class MobileDeviceAlreadyExists(Exception):
     pass
 
 
-class DeviceKeyQuerySet(models.QuerySet):
+class MobileDeviceQuerySet(SmartQuerySet):
 
-    def create_token(self, uuid, user, user_agent=''):
+    def activate_or_create(self, uuid, user, user_agent='', name=None, slug=None, is_primary=True):
         """
         This method must be called when user is authenticated.
-        It creates a new DeviceKey with auto generated token for the device and returns token.
-        If DeviceKey with same UUID exists DeviceKeyAlreadyExistsException is raised.
+        It creates a new MobileDevice with auto generated token for the device and returns token.
+        If MobileDevice with same UUID exists MobileDeviceAlreadyExists is raised.
         """
-        token = generate_key(length=64)
-        is_created_new_device_key = self.get_or_create(
-            uuid=uuid, user=user,
-            defaults={
-                'login_token': make_password(token),
-                'user_agent': user_agent[:256],
-                'is_active': True
-            }
-        )[1]
-        if is_created_new_device_key:
-            return token
+        secret_password = generate_key(length=64)
+        defaults = {
+            'login_token': make_password(secret_password),
```

Comment:
because for the token I have only one value to get and check token/OTP. Here I have UUID which identifies device and token. Use make_password is much more secure because every time you will call this method the password will be different (salt). So I firsly find device through UUID and now I can check password...